test/JDBC/expected/psql_logical_babelfish_db.out

```diff
@@ -311,6 +311,441 @@ go
+
+
+SELECT * FROM dms_db_babel_5077_dbo.dms_db_babel_5077_t1;
+GO
+~~START~~
+text#!#"sys"."varchar"
+NULL#!#chinese
+~~END~~
+
+
+UPDATE dms_db_babel_5077_dbo.dms_db_babel_5077_t2 SET dms_db_babel_5077_t2_col1 = 'collation' where dms_db_babel_5077_t2_col2 = 'estonian';
+GO
+~~ROW COUNT: 1~~
+
+
+UPDATE dms_db_babel_5077_dbo.dms_db_babel_5077_t2 SET dms_db_babel_5077_t2_col1 = 'NULL' where dms_db_babel_5077_t2_col2 = 'estonian';
+GO
+~~ROW COUNT: 1~~
+
+
+SELECT * FROM dms_db_babel_5077_dbo.dms_db_babel_5077_t2;
+GO
+~~START~~
+text#!#"sys"."varchar"
+NULL#!#estonian
+~~END~~
+
+
+UPDATE dms_db_babel_5077_dbo.dms_db_babel_5077_t3 SET dms_db_babel_5077_t3_col1 = 'collation' where dms_db_babel_5077_t3_col2 = 'chinese_text';
+GO
+~~ROW COUNT: 1~~
+
+
+UPDATE dms_db_babel_5077_dbo.dms_db_babel_5077_t3 SET dms_db_babel_5077_t3_col1 = 'NULL' where dms_db_babel_5077_t3_col2 = 'chinese_text';
+GO
+~~ROW COUNT: 1~~
+
+
+SELECT * FROM dms_db_babel_5077_dbo.dms_db_babel_5077_t3;
+GO
+~~START~~
+text#!#text
+NULL#!#chinese_text
+~~END~~
+
+
+UPDATE dms_db_babel_5077_dbo.dms_db_babel_5077_t4 SET dms_db_babel_5077_t4_col1 = 'collation' where dms_db_babel_5077_t4_col2 = 'estonian_varchar';
+GO
+~~ROW COUNT: 1~~
+
+
+UPDATE dms_db_babel_5077_dbo.dms_db_babel_5077_t4 SET dms_db_babel_5077_t4_col1 = 'NULL' where dms_db_babel_5077_t4_col2 = 'estonian_varchar';
+GO
+~~ROW COUNT: 1~~
+
+
+SELECT * FROM dms_db_babel_5077_dbo.dms_db_babel_5077_t4;
+GO
+~~START~~
+"sys"."varchar"#!#"sys"."varchar"
+NULL#!#estonian_varchar
+~~END~~
+
+
+UPDATE dms_db_babel_5077_dbo.dms_db_babel_5077_t5 SET dms_db_babel_5077_t5_col1 = 'collation' where dms_db_babel_5077_t5_col2 = 'default';
+GO
+~~ROW COUNT: 1~~
+
+
+UPDATE dms_db_babel_5077_dbo.dms_db_babel_5077_t5 SET dms_db_babel_5077_t5_col1 = 'NULL' where dms_db_babel_5077_t5_col2 = 'default';
+GO
+~~ROW COUNT: 1~~
+
+
+SELECT * FROM dms_db_babel_5077_dbo.dms_db_babel_5077_t5;
+GO
+~~START~~
+text#!#"sys"."varchar"
+NULL#!#default
+~~END~~
+
+
+UPDATE dms_db_babel_5077_dbo.dms_db_babel_5077_t6 SET dms_db_babel_5077_t6_col1 = 'collation' where dms_db_babel_5077_t6_col2 = 'unicode_cs_as';
+GO
+~~ROW COUNT: 1~~
+
+
+UPDATE dms_db_babel_5077_dbo.dms_db_babel_5077_t6 SET dms_db_babel_5077_t6_col1 = 'NULL' where dms_db_babel_5077_t6_col2 = 'unicode_cs_as';
+GO
+~~ROW COUNT: 1~~
+
+
+SELECT * FROM dms_db_babel_5077_dbo.dms_db_babel_5077_t6;
+GO
+~~START~~
+text#!#"sys"."varchar"
+NULL#!#unicode_cs_as
+~~END~~
+
+
+UPDATE dms_db_babel_5077_dbo.dms_db_babel_5077_t7 SET dms_db_babel_5077_t7_col1 = 'collation' where dms_db_babel_5077_t7_col2 = 'unicode_cs_as';
+GO
+~~ROW COUNT: 1~~
+
+
+UPDATE dms_db_babel_5077_dbo.dms_db_babel_5077_t7 SET dms_db_babel_5077_t7_col1 = 'NULL' where dms_db_babel_5077_t7_col2 = 'unicode_cs_as';
+GO
+~~ROW COUNT: 1~~
+
+
+SELECT * FROM dms_db_babel_5077_dbo.dms_db_babel_5077_t7;
+GO
+~~START~~
+text#!#"sys"."varchar"
+NULL#!#unicode_cs_as
+~~END~~
+
+
+-- tsql
+USE master;
+GO
+
+DROP DATABASE dms_db_babel_5077;
```

Comment:
just for completeness, can you add tests where predicate has the same column which is being updated and it also has PKEY. Cover the same for [n][var]char also

Reply:
updated in latest rev